serving/docs/lmi/conceptual_guide/lmi_engine.md

```diff
@@ -0,0 +1,72 @@
+# LMI running Engines
+
+In LMI, we offer two different running mode to operate the backend engine:
+- Distributed Environment (MPI): Used to operate on single machine multi-gpu or multiple machines multi-gpu use cases
+- Standard Python process (Python): Start a standalone python process to run the engine
+
+Depends on the engine architecture, settings and use case your are pursuing into, you can choose one of the option to run with LMI.
+Here we are providing the common Engine selection for different backends we offer:
+- TensorRT-LLM (MPI): Use multiple MPI processes to run the backends
+- LMI-Dist (MPI): Launching using multiple MPI processes to control for different GPUs
+- vLLM (Python): vLLM internally will use Ray to spin up multiple processes, so managed at vLLM layer
+- HuggingFace Accelerate (Python): HF Accelerate internally managed the process workflow
+- TransformerNeuronX (Python): The Neuron backend engine, internally it will use multi-threading to run with Neuron cores.
+
+In the next section, we will introduce a detailed breakdown on how we run those backends.
+
+## Python Engine operating in LMI (DJLServing)
+
+![python image](../imgs/python_mode.jpg)
+
+DJLServing could operate with Python with our Python Engine. In Python operating mode, we will spin up a python process
+from the system environment and allocate Accelerators(CPU/Neuron) for each processes through `CUDA_VISIBLE_DEVICES`. During auto-scaling mode,
+DJLServing could manage workers' Accelerators allocation and spin up process with different Accelerators (CPU/Neuron).
+Under python Engine mode, DJLServing will establish socket connection and talk to the python process.
+
+### Enablement
+
+You can use the following ways to enable MPI Engine:
+
+serving.properties
+
+```
+engine=Python
+```
+
+Environment variable
+
+```
+SERVING_LOAD_MODELS=test::Python=/opt/ml/model
```

Comment:
I don't think we want to promote this usage. We have to explain what `test` mean here.

In most cases, we are able to detect Python engine. We don't have to explicitly enable python engine. If really this is needed, we can introduce `SERVING_ENGINE` env var.

For MPI mode, we already have an env var: 'OPTION_MPI_MODE=true`

Reply:
For v9 we can rely just on HF_MODEL_ID (and optionally OPTION_ROLLING_BATCH if user wants to specify). But for v8, the auto Python engine mode logic doesn't really work if user specifies rolling batch but not engine.

Reply:
I'm wrong. In V8 the following are sufficient:
OPTION_MODEL_ID/HF_MODEL_ID
OPTION_ROLLING_BATCH
OPTION_MPI_MODE

we don't need SERVING_LOAD_MODELS even in v8